docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,286 @@
+
+[OSGi](https://osgi.org/specification/osgi.core/7.0.0/) specifies two annotations, [`@Attribute`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Attribute) and [`@Directive`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Directive), for this purpose. Any methods of the **custom bundle annotation** annotated with `@Attribute` or  `@Directive` will result in those becoming additional attributes or directives respectively of the resulting header clause _when a value is supplied_.
+
+##### `@Attribute`
+
+`@Attribute` allows you to add new or update existing attributes from the _bundle annotation_.
+
+```java
+@Capability(namespace = "foo.namespace")
+@interface Extended {
+	@Attribute("foo.attribute") // this attribute enhances the @Capability
+	String value();
+}
+
+// usage
+
+@Extended("bar")
+class Foo {}
+```
+
+which results in the manifest header:
+
+`Provide-Capability: foo.namespace;foo.attribute=bar`
+
+##### `@Directive`
+
+`@Directive` behaves similarly; with some caveats. You can add new or update existing directives for _non-specified namespaces_.
+
+```java
+@Capability(namespace = "foo.namespace")
+@interface Extended {
+	@Directive("foo.directive")
+	String value();
+}
+
+// usage
+
+@Extended("bar")
+class Foo {}
+```
+
+results in the manifest header:
+
+`Provide-Capability: foo.namespace;foo.directive:=bar`
+
+**However**, namespaces defined in the OSGi specifications will be validated and will not accept directives which are not part of the spec _unless_ they are prefixed with `x-`.
+
+```java
+@Capability(namespace = "osgi.extender", name = "bar", version = "1.0.0")
+@interface Extended {
+	@Directive("foo")
+	String value();
+}
+
+// usage
+
+@Extended("bar")
+public class Foo {}
+```
+
+will result in an error:
+
+`Unknown directive 'foo:' for namespace 'osgi.extender' in 'Provide-Capability'. Allowed directives are [effective:,uses:], and 'x-*'.`
+
+This next example however:
+
+```java
+@Capability(namespace = "osgi.extender", name = "bar", version = "1.0.0")
+@interface Extended {
+	@Directive("x-foo")
+	String value();
+}
+
+// usage
+
+@Extended("bar")
+public class Foo {}
+```
+
+results in the manifest header:
+
+`Provide-Capability: osgi.extender;osgi.extender=bar;version:Version="1.0.0";x-foo:=bar`
+
+**It should be noted** that default values for methods annotated with `@Attribute` and `@Directive` are deemed to be for documentation purposes only and will not be emitted into resulting headers.
+
+### Where to find Bundle Annotations
+
+**OSGi** *bundle annotations* can be found in the `osgi.annotation` bundle.
+
+```xml
+<dependency>
+  <groupId>org.osgi</groupId>
+  <artifactId>osgi.annotation</artifactId>
+  <version>7.0.0</version>
+</dependency>
+```
+
+**Bnd** *bundle annotations* can be found in `biz.aQute.bndlib` or `biz.aQute.bnd.annotations` bundles.
```

Comment:
I think we should ONLY direct people to use `biz.aQute.bnd.annotations` for Bnd annotations. bndlib is too much other stuff.